udata/frontend/markdown.py

```diff
@@ -39,21 +40,35 @@ def nofollow_callback(attrs, new=False):
     That callback is not splitted in order to parse the URL only once.
     """
 
```

Comment:
If we go down this road, we should rename this function: it's not handling only nofollow, but a bunch of different treatments. The docstring should also be updated. More comments in the function body would be nice too, to understand the different steps.